scripts/asm-installer/install_asm

```diff
@@ -151,6 +157,15 @@ main() {
     exit_if_stackdriver_not_enabled
   fi
 
+  if [[ "${CA}" = "mesh_ca" ]] && can_modify_at_all; then
+    if [[ "${ENABLE_HUB_IDNS}" -eq 1 ]]; then
+      get_hub_workload_identity
+      init_meshca_with_hub_idns
+    else
+      init_meshca
+    fi
+  fi
+
```

Comment:
1) I changed this section before you put up this PR, sorry! You'll have to rebase.

2) I think we can put this logic inside of init_meshconfig instead of here.